### PR TITLE
Provide the ability to create a new cli application registration

### PR DESCRIPTION
**Pull request by chkeita**
## Summary of the Pull Request

_What is this about?_

## PR Checklist
* [x] Applies to work item: #147 
* [ ] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/onefuzz) and sign the CLI.
* [ ] Tests added/passed
* [ ] Requires documentation to be updated
* [ ] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

## Info on Pull Request

_What does this include?_

## Validation Steps Performed

_How does someone test & validate?_

### CODE REVIEW

File: src/deployment/registration.py [hunk 1]
```diff
@@ -369,13 +414,32 @@ def main():
...
+
+# todo: update the internal docs (https://dev.azure.com/msresearch/OneFuzz/_git/OneFuzz?path=%2FREADME.md&_a=preview&version=GBinternal-information) with
+
+# Mandatory role assigment
+# if your org require that you prevent guest access by requiring a role assigment.
+# The default CLI will not be able to access the service.
+# you will need to create a new cli registration by executing the following
+# ```
+# python registration.py create_cli_registration INSTANCENAME [--registration_name REGISTRATIONNAME]
+# ```
```
Comment (chkeita) [inline src/deployment/registration.py:445]: Reminder to update the documentation

Comment (chkeita) [reply src/deployment/registration.py]: will update the docs once this PR is approved 

### Guest account access
If you need [prevent guest access](https://osgwiki.com/wiki/Guest_Accounts_in_Azure#2._Prevent_users_not_assigned_to_your_app_from_accessing_your_application.2Fservice) by requiring a role assigment, the default CLI will not be able to access the service.
You will need to create a new cli registration by executing the following.
```
python registration.py create_cli_registration INSTANCENAME [--registration_name REGISTRATIONNAME]
```

Then use the provided credentials in your onefuzz config 
```
onefuzz config --client_id CLIENTID
```

**Comment by bmc-msft** [comment]
Performing integration tests

**Review by bmc-msft — CHANGES_REQUESTED** [review]
Currently fails due to:

```
running deployment
Traceback (most recent call last):
  File "deploy.py", line 63, in <module>
    from registration import (
ImportError: cannot import name 'update_pool_registration' from 'registration' (/tmp/tmpitt7aqre/registration.py)
Resource group 'pr-check-3ff769f69f004a0780e6d3a3cbc7253c' could not be found.
Traceback (most recent call last):
  File "/home/bcaswell/projects/onefuzz/tools/check-pr", line 254, in <module>
    main()
  File "/home/bcaswell/projects/onefuzz/tools/check-pr", line 247, in main
    d.run(merge_on_success=args.merge_on_success)
  File "/home/bcaswell/projects/onefuzz/tools/check-pr", line 217, in run
    self.deploy(dist)
  File "/home/bcaswell/projects/onefuzz/tools/check-pr", line 168, in deploy
    subprocess.check_call(cmd, shell=True)
  File "/usr/lib/python3.8/subprocess.py", line 364, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command './deploy-venv/bin/python deploy.py uksouth pr-check-3ff769f69f004a0780e6d3a3cbc7253c pr-check-3ff769f69f004a0780e6d3a3cbc7253c cicd' returned non-zero exit status 1.
```

**Comment by bmc-msft** [comment]
Integration testing in progress (currently waiting on fresh build)

**Review by bmc-msft — APPROVED** [review]

**Pull request merged**